corehq/apps/userreports/README.md

```diff
@@ -1288,6 +1288,7 @@ Example assuming "village" is a location ID, which is converted to names using t
   "choice_provider": {
       "type": "location",
       "include_descendants": true,
+      "order_by_hierarchy": true,
```

Comment:
Looks like i'm a year late on asking this, but why isn't this the default? https://github.com/dimagi/commcare-hq/pull/15017

Reply:
I got pushback on changing the behavior for existing reports when I added the "show_full_path" option and settled on adding it to the README as a "sort-of default": https://github.com/dimagi/commcare-hq/pull/14309
This option is probably less objectionable, but someone did explicitly build it the old way at some point.  I do think there are only two reasonable configurations here:

`show_full_path` and `order_by_hierarchy` are both true, and the query uses `filter_path_by_user_input`.

`show_full_path` and `order_by_hierarchy` are both false, and the query uses `filter_by_user_input` (not currently possible).

That is, the current defaults are pretty nonsensical - If only the name is displayed, then sorting it alphabetically makes sense, but I don't think it makes sense for a user to type "Massachusetts" and see "Boston" in the results unless it's listed as "Massachusetts/Suffolk/Boston".

Reply:
I agree with your thoughts on reasonable defaults. Could we make the change to only use `show_full_path` and deprecate `order_by_hierarchy`? (this pr probalby wouldn't be the place to do it)

Reply:
Hmm, yeah, that sounds reasonable to me.  Think it's also fair to switch the filtering logic if you don't specify `show_full_path = True`?  I could see that getting some pushback.

One thing I've found from talking to people using locations on larger projects is that they sometimes try to reimplement the hierarchical nature of locations with project-specific patterns.  For instance, ICDS has numerical site codes defined in terms of their ancestry (for instance, Massachusetts would be 123, Suffolk would be 12312, and Boston would be 1231287).  I've also heard of projects where "Boston" would have the name "Massachusetts - Suffolk - Boston".